### PR TITLE
Migrate `MLFLOW_EXPERIMENT_ID` and `MLFLOW_EXPERIMENT_NAME`

### PR DESCRIPTION
**Pull request by gabrielfu**
## Related Issues/PRs

#6379 #8849 

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: tests/tracking/fluent/test_fluent.py [hunk 1]
```diff
@@ -62,20 +64,20 @@ def __init__(self):
 
     @classmethod
     def assert_values(cls, exp_id, name):
-        assert os.environ.get(_EXPERIMENT_NAME_ENV_VAR) == name
-        assert os.environ.get(_EXPERIMENT_ID_ENV_VAR) == exp_id
+        assert MLFLOW_EXPERIMENT_NAME.get() == name
```
Comment (harupy) [inline tests/tracking/fluent/test_fluent.py:67]: Is it possible to remove `HelperEnv` in a follow-up PR?

Comment (harupy) [reply tests/tracking/fluent/test_fluent.py:67]: We can directly use:

```python
assert MLFLOW_EXPERIMENT_NAME.get() == name
assert MLFLOW_EXPERIMENT_ID.get() == exp_id
```

and:

```python
monkeypatch.{setenv|devenv}
```

File: mlflow/tracking/fluent.py [hunk 2]
```diff
@@ -1767,14 +1769,14 @@ def _get_or_start_run():
 
 
 def _get_experiment_id_from_env():
-    experiment_name = env.get_env(_EXPERIMENT_NAME_ENV_VAR)
-    experiment_id = env.get_env(_EXPERIMENT_ID_ENV_VAR)
```
Comment (harupy) [inline mlflow/tracking/fluent.py:1771]: Is `env` used anywhere else? If not, let's remove it.

File: tests/tracking/fluent/test_fluent.py (hunk 1 above)
Comment (gabrielfu) [reply tests/tracking/fluent/test_fluent.py:67]: monkeypatch does sound like a better option, I'll create a follow up PR

File: mlflow/tracking/fluent.py (hunk 2 above)
Comment (gabrielfu) [reply mlflow/tracking/fluent.py:1771]: looks like it's not being used anywhere else, and I forgot to remove the import :)

**Pull request merged**